### ISSUE TITLE
Margin in NCA loss

### ISSUE THREAD

**Issue by EvelynChee** [arthurdouillard/incremental_learning.pytorch#59]
https://github.com/arthurdouillard/incremental_learning.pytorch/blob/0d25c2e12bde4a4a25f81d5e316751c90e6f789b/inclearn/lib/losses/base.py#L72-L74

Based on the equation in the paper, shouldn't line 74 be `similarities - margins` instead? Currently it seems to remove the margin for all classes instead of the only the target class?

**Comment by arthurdouillard** [comment]
Line 73 does that, `margins` is zero for all negative class, and of value `margin` for the target class.

**Comment by EvelynChee** [comment]
@arthurdouillard  yes, `margins` has that but in line 74, the value `margin` is deducted instead of `margins`?

**Comment by arthurdouillard** [comment]
Oh yes, you're definitely right!

Thanks for catching that, I'll fix it.
